lib/util/request.js

```diff
@@ -1,31 +1,57 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
+* License, v. 2.0. If a copy of the MPL was not distributed with this file,
+* You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 'use strict'
 
 const urlFormat = require('url').format
 const underscore = require('underscore')
 const responses = require('../data/responses')
+const reconcileResponses = require('../data/contribution-responses')
 const urlParse = require('../../browser-laptop/app/common/urlParse')
 
 const hostLookup = {
   'ledger.brave.com': 'ledger-proxy.privateinternetaccess.com',
-  'ledger.mercury.basicattentiontoken.org': 'mercury-proxy.privateinternetaccess.com'
+  'ledger-staging.mercury.basicattentiontoken.org': 'ledger-staging.mercury.basicattentiontoken.org' // 'mercury-proxy.privateinternetaccess.com'
 }
 
-const getResponse = (url, currWallet) => {
+const getResponse = (url, currWallet, httpMethod) => {
   let response = {}
   const path = urlParse(url).path
-
-  if (path.match(/\/v2\/wallet\?publicKey/)) {
-    return responses['/v2/wallet/id'][currWallet]
-  } else if (path.match(/\/v2\/wallet/)) {
-    return responses['/v2/wallet/balance'][currWallet]
-  } else if (path.match(/\/v2\/registrar\/persona/)) {
-    return path.split('/').length === 4
-      ? responses['/v2/registrar/persona']['initial-registrar']
-      : {}
+  if (httpMethod === 'GET') {
```

Comment:
Can we change some of these checks from a regex match to a key lookup in the response object? (Where the path is the key) some of these are 1:1

Reply:
I was able to make a few of these into key lookups. The rest of them come in as UUID/GUID and don't lend themselves for key lookups.

Reply:
Great! :D